### ISSUE TITLE
 ResideInNamespace contains instead of exact

### ISSUE THREAD

**Issue by jezzsantos** [TNG/ArchUnitNET#126]
Turns out that the rule `ResideInNamespace("anamespace", false)` does a partial match on the namespace instead of a full match, and so extra types are captured from other namespaces that should not be, and this makes no sense logically.

For example: I have assemblies with these namespaces:
`Api.Interfaces` and `External.Api.Interfaces`.

If use this rule: 
```
var responseDtos = ArchRuleDefinition.Types(true).That().ArePublic().As("Things")
                .And().ResideInNamespace("Api.Interfaces")
```
I process types that are in both the `Api.Interfaces` and `External.Api.Interfaces` namespaces, which is not what I would expect at all.
 
Looking at the code, it appears that `ResideInNamespace` calls `NamingExtensions.FullNameContains` which does a partial match on the namespace.

**Comment by brandhuf** [comment]
This behaviour is changed in #145

**Comment by fgather** [comment]
This will be a breaking change. We will release it soon, but will try to merge some other breaking changes as well, while we're at it.

**Comment by fgather** [comment]
Fix released in 0.10.0
